pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -17,16 +17,23 @@ const (
 )
 
 // Condition
+// Type - The condition type.
+// Status - The condition status.
+// Reason - The reason for the condition.
+// Message - The human readable description of the condition.
+// staged - The condition is staged for commit.
 type Condition struct {
 	Type               string      `json:"type"`
 	Status             string      `json:"status"`
 	Reason             string      `json:"reason,omitempty"`
 	Message            string      `json:"message,omitempty"`
 	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	staged             bool
 }
 
 // Update this condition with another's fields.
 func (r *Condition) Update(other Condition) {
+	r.staged = true
```

Comment:
So now each condition has a `staged` bool attached to it, and `staged` gets set to true if a Condition changes.
I noticed you didn't include JSON serialization info and made `staged` lowercase. Does this keep it from showing up on the resource?

Reply:
I modeled this after `Git`.  All conditions are un-staged by default.  A condition is _staged_  when explicitly set (or updated) using `SetCondition()` like `git add`.  Then, calling `CommitConditions()` removes the un-staged conditions before the resource is stored.  This way, the validation code does not need to explicitly delete condition but rather only set the onces that still exist.

Correct, `staged` is not included in the JSON serialization.